RecoPixelVertexing/PixelTrackFitting/src/PixelFitterByHelixProjections.cc

```diff
@@ -104,36 +104,20 @@ reco::Track* PixelFitterByHelixProjections::run(
   int nhits = hits.size();
   if (nhits <2) return 0;
 
-  vector<GlobalPoint> points(nhits);
-  vector<GlobalError> errors(nhits);
-  vector<bool> isBarrel(nhits);
-  
-  if (theTrackerWatcher.check(es)) {
-    edm::ESHandle<TrackerGeometry> trackerESH;
-    es.get<TrackerDigiGeometryRecord>().get(trackerESH);
-    theTracker = trackerESH.product();
-  }
-
-  if (theFieldWatcher.check(es)) {
+  if (!theField) { 
     edm::ESHandle<MagneticField> fieldESH;
     es.get<IdealMagneticFieldRecord>().get(fieldESH);
     theField = fieldESH.product();
```

Comment:
If a mixture of magnetic field values in a single job ought to be supported, we have a problem also in https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/RecoTracker/TkMSParametrization/interface/PixelRecoUtilities.h#L27 (lesser though as only the value at the center is cached, so no crashes but incorrect value gets used).

Here is a list of affected files
```
$ git grep --name-only PixelRecoUtilities::
RecoMuon/TrackerSeedGenerator/src/L1MuonPixelTrackFitter.cc
RecoPixelVertexing/PixelLowPtUtilities/src/TrackFitter.cc
RecoPixelVertexing/PixelTrackFitting/src/KFBasedPixelFitter.cc
RecoPixelVertexing/PixelTrackFitting/src/PixelFitterByConformalMappingAndLine.cc
RecoPixelVertexing/PixelTrackFitting/src/PixelFitterByHelixProjections.cc
RecoPixelVertexing/PixelTriplets/plugins/PixelTripletHLTGenerator.cc
RecoPixelVertexing/PixelTriplets/plugins/PixelTripletLargeTipGenerator.cc
RecoPixelVertexing/PixelTriplets/plugins/PixelTripletNoTipGenerator.cc
RecoTracker/ConversionSeedGenerators/plugins/HitPairGeneratorFromLayerPairForPhotonConversion.cc
RecoTracker/TkHitPairs/src/InnerDeltaPhi.cc
RecoTracker/TkMSParametrization/interface/LongitudinalBendingCorrection.h
RecoTracker/TkSeedGenerator/plugins/MultiHitGeneratorFromChi2.cc
```